driver-reactive-streams/src/main/com/mongodb/reactivestreams/client/BatchCursorPublisher.java

```diff
@@ -14,10 +14,13 @@
  * limitations under the License.
  */
 
-package com.mongodb.internal.async.client.gridfs;
+package com.mongodb.reactivestreams.client;
 
-import com.mongodb.internal.async.SingleResultCallback;
-
-interface GridFSIndexCheck {
-    void checkAndCreateIndex(SingleResultCallback<Void> callback);
+/**
+ * A publisher for batch cursors
+ *
+ * @param <T> The type of documents the cursor contains
+ * @since 4.2
+ */
+public interface BatchCursorPublisher<T> extends BatchCursorPublisherGeneric<BatchCursor<T>, T> {
```

Comment:
Why is it necessary to make all of these batch cursor publishers?  Is that a requirement to make this work properly, or something extra?

Reply:
Maybe for post batch resume token support?

Reply:
Yeah, I created 3, so that only the base one would require two generics.  The AggregateBatchCursorPublisher just provides the AggregateBatchCursor which has the resume token support.

Happy to rename the base interface if the naming is confusing. Or remove the base one and make each implementation explicitly declare their BatchCursor type.

Reply:
I was more wondering why there is a functional requirement for all of these to publish batch cursors.

Reply:
We need them internally, but they don't need to be made public - have updated.